appcues/src/main/java/com/appcues/ui/primitive/ImagePrimitive.kt

```diff
@@ -9,18 +11,18 @@ import com.appcues.ui.LocalImageLoader
 import com.appcues.ui.extensions.blurHashPlaceholder
 import com.appcues.ui.extensions.getImageLoader
 import com.appcues.ui.extensions.getImageRequest
-import com.appcues.ui.extensions.styleImageAspect
+import com.appcues.ui.extensions.imageAspectRatio
 import com.appcues.ui.extensions.styleSize
 import com.appcues.ui.extensions.toImageAsyncContentScale
 import com.appcues.ui.utils.rememberBlurHashDecoded
 
 @Composable
-internal fun ImagePrimitive.Compose(modifier: Modifier) {
+internal fun ImagePrimitive.Compose(modifier: Modifier, matchParentBox: BoxScope? = null) {
     val context = LocalContext.current
     val decodedBlurHash = rememberBlurHashDecoded(blurHash = blurHash)
 
     AsyncImage(
-        modifier = modifier.then(Modifier.styleSize(style, contentMode).styleImageAspect(this)),
+        modifier = modifier.applyImageModifier(this, matchParentBox),
```

Comment:
image is the only primitive that uses styleSize twice, this is why we are passing the BoxScope as parameter